test/dbpoolTest.js

```diff
@@ -216,4 +217,46 @@ describe('DBPool Class Tests', () => {
         });
       });
   });
+  describe('new connection event', () => {
+    it('executes a callback when new connection event occurs', (done) => {
+      const pool = new DBPool({ url: '*LOCAL' });
+      // register the listener
+      pool.on('new connection', (connection) => {
+        expect(connection).to.be.instanceof(Connection);
+        done();
+      });
+      pool.createConnection();
+    });
+
+    it('sets up library list for every new connection', (done) => {
+      const pool = new DBPool({ url: '*LOCAL' }, { incrementSize: 3 });
+      let counter = 0;
+
+      async function checkLibList() {
+        for (let i = 0; i < pool.connections.length; i += 1) {
+        // eslint-disable-next-line no-await-in-loop
+          const result = await pool.connections[i].getStatement().exec("SELECT SCHEMA_NAME from qsys2.library_list_info WHERE TYPE = 'USER'");
+          expect(result).to.be.a('array');
+          expect(result[0]).to.be.a('object');
+          expect(result[0].SCHEMA_NAME).to.equal('QXMLSERV');
+          expect(result[1]).to.be.a('object');
+          expect(result[1].SCHEMA_NAME).to.equal('QGPL');
+          expect(result[2]).to.be.a('object');
+          expect(result[2].SCHEMA_NAME).to.equal('QIWS');
+        }
+        done();
+      }
+
+      pool.on('new connection', async (connection) => {
+        await connection.setLibraryList(['QXMLSERV', 'QGPL', 'QIWS']);
+        counter += 1;
+        if (counter === 3) {
+          checkLibList();
+        }
```

Comment:
While ostensibly the reason for this feature to exist is to set the library list, all we need to do to test this feature is verify that the callback was called and was passed a connection object. This test would be much simpler if the callback just set some attribute on the connection object then verified it existed on each item in the pool.

Reply:
> all we need to do to test this feature is verify that the callback was called and was passed a connection object. 

That is what the [first test](https://github.com/IBM/nodejs-idb-pconnector/pull/76/files#diff-40903a9739dc9b5cb1f1bc0a0e8e2809R220) does.

Here we verify that the library is really set properly.

Reply:
If we already have a test to check that the callback works, then I don't find much value in this test. The feature is for allowing callbacks to be called when a connection is created. The implementation details of the connection object are outside the scope of this test.